bluesky_ui/widgets.py

```diff
@@ -17,6 +21,298 @@ def merge_parameters(widget_iter):
             if w.isEnabled()}
 
 
+class UniqueValidator(QValidator):
+    '''A ``QValidator`` that ensures that the string is not in invalid_list.'''
+
+    def __init__(self, invalid_list, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.invalid_list = invalid_list
+
+    # 'validate' function that checks uniqueness
+    def validate(self, inputStr, pos):
+        if inputStr in self.invalid_list:
+            return (self.Invalid, inputStr, pos)
+        else:
+            return (self.Acceptable, inputStr, pos)
+
+    def fixup(self, text):
+        '''Modifes 'text' in-situ in order to attempt to make it unique'''
+        # TODO : fill out this method, an example can be found at
+        # https://stackoverflow.com/questions/34055174/qvalidator-fixup-issue
+
+
+def table_unique_mtest_factory(name, parent, table):
```

Comment:
I think it should be named as `table_unique_mtext_factory` (note `mtest` --> `mtext` change), and the corresponding change should be made in places where it's used.

Reply:
fixed, thanks for the careful review @mrakitin